src/kernel/stage3.c

```diff
@@ -57,7 +57,7 @@ static buffer bulk_test_buffer(heap h)
 {
     buffer b = allocate_buffer(h, BULK_TEST_BUFSIZ);
     for (int i = 0; i < (BULK_TEST_BUFSIZ / 10); i += 8) {
-        bprintf(b, "%8d %8d %8d %8d %8d %8d %8d %8d\r\n",
+        bprintf(b, "%08d %08d %80d %08d %08d %08d %08d %08d\r\n",
```

Comment:
In the format string there is a `%80d` instead of `%08d`.
I also found a few more places where we need to change the flags so that the resulting string is padded with '0' instead of ' ':
- src/unix/mmap.c, do_demand_page(): `0x%16lx` should be `0x%016lx`, or better yet `%p`
- src/xen/xennet.c, xennet_attach(): `%2x:%2x:%2x:%2x:%2x:%2x` should be `%02x:%02x:%02x:%02x:%02x:%02x`
- src/tfs/tfs.c, report_sha256(): `%16lx%16lx%16lx%16lx` should be `%016lx%016lx%016lx%016lx`
- platform/pc/boot/stage2.c, kernel_read_complete(): `%8lx` should be `%08lx`

Reply:
Good catch, I did miss all the hex formatted stuff. Thanks!